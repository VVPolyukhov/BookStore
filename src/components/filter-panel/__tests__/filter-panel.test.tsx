import React from 'react'
import { shallow } from 'enzyme'

import FilterPanel, { PropsType } from '../filter-panel'

describe('FilterPanel:', () => {
    const props: PropsType = {
        filterBy: 'all',
        setFilter: () => {}
    }

    describe('Button is disabled?', () => {
        const Wrapper = shallow(<FilterPanel {...props} />)

        it('renders properly', () => {
            expect(Wrapper).toMatchSnapshot()
        })
        it('button is disabled if (filterBy == ("all" | null)', () => {
            const button = Wrapper.find('Button')
            expect(button.prop('disabled')).toBeTruthy()
        })
    })
    
    describe('Button is available?', () => {
        const nextProps = {
                ...props,
                filterBy: 'author'
            }
        const Wrapper = shallow(<FilterPanel {...nextProps} />)

        it('renders properly', () => {
            expect(Wrapper).toMatchSnapshot()
        })
        it('button is available if (filterBy != "all")', () => {
            const button = Wrapper.find('Button')
            expect(button.prop('disabled')).toBeFalsy()
        })
    })
})