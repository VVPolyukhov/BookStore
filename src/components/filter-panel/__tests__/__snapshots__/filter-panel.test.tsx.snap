// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel: Button is available? renders properly 1`] = `
<Fragment>
  <Menu
    fluid={true}
    tabular={true}
    text={true}
    vertical={true}
  >
    <MenuItem
      header={true}
    >
      Сортировать по:
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Популярности
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Цене (сначала дешевые)
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Цене (сначала дорогие)
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Названию книги
    </MenuItem>
    <MenuItem
      active={true}
      onClick={[Function]}
    >
      Автору
    </MenuItem>
  </Menu>
  <Button
    as="button"
    disabled={false}
    onClick={[Function]}
    size="small"
  >
    Отменить сортировку
  </Button>
</Fragment>
`;

exports[`FilterPanel: Button is disabled? renders properly 1`] = `
<Fragment>
  <Menu
    fluid={true}
    tabular={true}
    text={true}
    vertical={true}
  >
    <MenuItem
      header={true}
    >
      Сортировать по:
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Популярности
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Цене (сначала дешевые)
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Цене (сначала дорогие)
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Названию книги
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      Автору
    </MenuItem>
  </Menu>
  <Button
    as="button"
    disabled={true}
    onClick={[Function]}
    size="small"
  >
    Отменить сортировку
  </Button>
</Fragment>
`;
